### PR TITLE
fix: better None protection when tainting a grpc message

### PR DESCRIPTION
**Pull request by juanjux**
## Checklist

- [X] Change(s) are motivated and described in the PR description
- [X] Testing strategy is described if automated tests are not included in the PR
- [X] Risks are described (performance impact, potential for breakage, maintainability)
- [X] Change is maintainable (easy to change, telemetry, documentation)
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [X] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/contrib/grpc/client_interceptor.py [hunk 1]
```diff
@@ -173,8 +173,10 @@ def __next__(self):
                 "grpc.response_message",
                 (n,),
             )
-            if result and "response" in result:
-                n = result["response"].value
+            if result:
```
Comment (brettlangdon) [inline ddtrace/contrib/grpc/client_interceptor.py:176]: do we need this on L88 as well?

File: releasenotes/notes/asm-gprc-not-none-788b4b435b931a11.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+fixes:
+  - |
+    ASM: protect against potentially returning `None` when tainting a gRPC message.
```
Comment (brettlangdon) [inline releasenotes/notes/asm-gprc-not-none-788b4b435b931a11.yaml:3]:
```suggestion
    ASM: protect against potentially returning ``None`` when tainting a gRPC message.
```

File: ddtrace/contrib/grpc/client_interceptor.py (hunk 1 above)
Comment (juanjux) [reply ddtrace/contrib/grpc/client_interceptor.py:176]: Oh yes, thanks, I will make the change

**Review by brettlangdon — COMMENTED** [review]
is there a regression test we can add? do we know when/how this can happen?

**Comment by juanjux** [comment]
> is there a regression test we can add? do we know when/how this can happen?

Not with the current code since we don't even change the `response` but the `type(response)`.

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**